src/main/scala/org/opennetworkinsight/dns/DNSPreLDA.scala

```diff
@@ -93,116 +78,57 @@ object DNSPreLDA {
         */
       val lines = Source.fromFile(feedbackFile).getLines().toArray.drop(1)
       val feedback: RDD[String] = sc.parallelize(lines)
-      val feedbackDataFrame = feedback.map(_.split(",")).map(row => Feedback(row(FrameTimeIndex),
+      feedback.map(_.split(","))
+        .filter(row => row(DnsSevIndex).trim.toInt == 3)
+        .map(row => Feedback(row(FrameTimeIndex),
         row(UnixTimeStampIndex),
         row(FrameLenIndex).trim.toInt,
         row(IpDstIndex),
         row(DnsQryNameIndex),
         row(DnsQryClassIndex),
         row(DnsQryTypeIndex),
         row(DnsQryRcodeIndex),
-        row(DnsSevIndex).trim.toInt)).toDF()
-      val result = feedbackDataFrame.filter("dnsSev = 3").select("frameTime", "unixTimeStamp", "frameLen", "ipDst",
-        "dnsQryName", "dnsQryClass", "dnsQryType", "dnsQryRcode").map(_.mkString(","))
-      val toDuplicate: org.apache.spark.rdd.RDD[String] = result.flatMap(x => List.fill(duplicationFactor)(x))
-      toDuplicate
+        row(DnsSevIndex).trim.toInt))
+        .flatMap(row => List.fill(duplicationFactor)(row))
+        .toDF()
+        .select("frameTime", "unixTimeStamp", "frameLen", "ipDst",
```

Comment:
is there a reason why these columns are named with absolute constants and not the names in the Schema object ?